Model/Service/ReportNewRules.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace Flancer32\Csp\Model\Service;
+
+use Flancer32\Csp\Api\Data\Fl32RuleSentInterfaceFactory;
+use Flancer32\Csp\Api\Fl32RuleSentRepositoryInterface;
+use Flancer32\Csp\Api\Repo\Data\Rule as ERule;
+use Magento\Framework\App\Helper\Context;
+use Magento\Framework\DataObject;
+use Psr\Log\LoggerInterface;
+
+class ReportNewRules
+{
+    //Uses consecutive enumeration of ids to determine last sent
+    //scheme: last sent -> read all successors -> prepare mail -> send -> insert entries into reports sent
+    /**
+     * @var HtmlFormatter
+     */
+    private $htmlFormatter;
+    /**
+     * @var DatabaseRule
+     */
+    private $ruleDataService;
+    /**
+     * @var Fl32RuleSentRepositoryInterface
+     */
+    private $fl32RuleSentRepository;
+    /**
+     * @var EmailSend
+     */
+    private $send;
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+    /**
+     * @var Fl32RuleSentInterfaceFactory
+     */
+    private $fl32RuleSentFactory;
+
+    public function __construct(
+        DatabaseRule $ruleDataService,
+        HtmlFormatter $htmlFormatter,
+        Fl32RuleSentRepositoryInterface $fl32RuleSentRepository,
+        Fl32RuleSentInterfaceFactory $fl32RuleSentFactory,
+        EmailSend $send,
+        Context $context)
+    {
+        $this->ruleDataService = $ruleDataService;
+        $this->htmlFormatter = $htmlFormatter;
+        $this->fl32RuleSentRepository = $fl32RuleSentRepository;
+        $this->send = $send;
+        $this->logger = $context->getLogger();
+        $this->fl32RuleSentFactory = $fl32RuleSentFactory;
+    }
+
+    /**
+     * @throws \Magento\Framework\Exception\LocalizedException
+     * @throws \Exception
+     */
+    public function execute(bool $throwException = false)
+    {
+        $rulesToSend = $this->ruleDataService->getRulesToSend();
+
+        if (!count($rulesToSend)) {
+            return;
+        }
+
+        $reportHtmlRules = $this->htmlFormatter->renderNewRulesSection($rulesToSend);
+
+        try {
+            $recepients = $this->send->with($reportHtmlRules);
+            $this->updateProtocolTables($rulesToSend, $recepients);
+        } catch (\Exception $exception) {
+            $this->logger->emergency($exception->getMessage());
+            if ($throwException) {
+                throw $exception;
+            }
+        }
+    }
+
+    /**
+     * extendable with new for-each blocks and parameters with different dataobject implementations
```

Comment:
what does this mean? can you give an example?

Reply:
I case any other resource model would be added, it would require another foreach, since the for each block would be bound to a particular dataobject model.
This was the case with the removed model for reporting the reports

Reply:
got it, then better remove the comment